cartographer/mapping/id.h

```diff
@@ -322,6 +323,43 @@ class MapById {
 
   bool empty() const { return begin() == end(); }
 
+  ConstIterator lower_bound(int trajectory_id, const common::Time& time) {
```

Comment:
Also, this does not follow the semantic set by the STL: 

> Returns an iterator pointing to the first element in the range [first, last) that is not less than (i.e. greater or equal to) value, or last if no such element is found.

Reply:
It adds the idea that you have data grouped by trajectories and also uses the time() property for ordering rather than the keys. Should we rename it?